### PR TITLE
Isolate EnigmaStruct family of structures.

### LINKED ISSUE #1593: "#define byte" macro in javastruct.h conflicting with google protobuf headers
https://github.com/enigma-dev/enigma-dev/blob/d0487e73f36b51de9f9f9fbbf4d62d73fa4a9043/CommandLine/emake/EnigmaCallbacks.hpp#L5
Move #include "backend/JavaCallbacks.h"  to line 10 (after #include "codegen/server.pb.h")

### PR DESCRIPTION
**Pull request by JoshDreamland**
Move them to namespace `deprecated::JavaStruct` where they won't interfere with protos and other libraries.

Fixes #1594.
Fixes #1593.

### CODE REVIEW

**Comment by RobertBColton** [comment]
What fundies said has some truth to it. emake is the only place in our stuff using EnigmaStruct now, for an empty game. Changing it would speed up running empty games from emake since it wouldn't have to go through the conversion to GameData. Our continuous integration happens to run empty games a lot, so it would maybe shave a few minutes off the total run time of our CI builds.

File: CommandLine/emake/Game.hpp [hunk 1]
```diff
@@ -12,30 +12,30 @@ class Game
...
+  std::vector<deprecated::JavaStruct::Script> _scripts;
+  std::vector<deprecated::JavaStruct::Sprite> _sprites;
+  std::vector<deprecated::JavaStruct::Shader> _shaders;
+  std::vector<deprecated::JavaStruct::Sound> _sounds;
+  std::vector<deprecated::JavaStruct::Font> _fonts;
+  std::vector<deprecated::JavaStruct::Background> _backgrounds;
+  std::vector<deprecated::JavaStruct::Path> _paths;
+  std::vector<deprecated::JavaStruct::Timeline> _timelines;
+  std::vector<deprecated::JavaStruct::GmObject> _objects;
+  std::vector<deprecated::JavaStruct::Room> _rooms;
```
Comment (RobertBColton) [inline CommandLine/emake/Game.hpp]: Let me take this opportunity to point out we should either make it possible to compile with the deprecated namespace removed completely (non-existant in the source) and or you could just add a using here.

File: CompilerSource/backend/GameData.cpp [hunk 2]
```diff
@@ -306,7 +306,7 @@ ObjectData::ObjectData(const ::GmObject &object, const ESLookup &lookup):
 
 RoomData::RoomData(const buffers::resources::Room &q, const std::string& name):
   BaseProtoWrapper(q), name(name) {}
-RoomData::RoomData(const ::Room &room, const ESLookup &lookup):
+RoomData::RoomData(const deprecated::JavaStruct::Room &room, const ESLookup &lookup):
```
Comment (RobertBColton) [inline CompilerSource/backend/GameData.cpp]: Again, could use a using here. I mean this is a source and not a header so doing it here wouldn't be as bad.

**Comment by JoshDreamland** [comment]
I'll assume he meant Proto, not `GameData`. And for the most part, it is. As you can see, it's only currently using EnigmaStruct as a storage mechanism for SOG, which I am removing locally. For now, this can submit as-is.

File: CommandLine/emake/Game.hpp (hunk 1 above)
Comment (JoshDreamland) [reply CommandLine/emake/Game.hpp]: We deprecate things when we want them to go away soon. Deprecation means new code should not use something. In the case of EnigmaStruct, it's not going away until LGM supports Proto, which I'm guessing isn't anytime soon. I will remove references to it in existing code, as possible, in follow-up PRs.

File: CompilerSource/backend/GameData.cpp (hunk 2 above)
Comment (JoshDreamland) [reply CompilerSource/backend/GameData.cpp]: I'm happy to leave it explicit for the remainder of its life.

**Pull request merged**